modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -262,13 +267,19 @@ public Object visit(PropertyIsLessThanOrEqualTo filter, Object extraData) {
     @Override
     public Object visit(PropertyIsLike filter, Object extraData) {
         BasicDBObject output = asDBObject(extraData);
-        String expr = convert(filter.accept(this, null), String.class);
+        Expression filterExpression = filter.getExpression();
+        // Mongo's $regex operator only works on fields
+        if (!(filterExpression instanceof PropertyName)) {
+            throw new UnsupportedOperationException();
+        }
```

Comment:
This restrictions should be communicated in the UnsuppoertedOperationMessage....

And the splitting visitor need to check this restriction and only ask MongoDB to encode like expressions using a property name.

I am going to go hack up an example, but the MongoDBDataStore uses deprecated PostPreProcessFilterSplittingVisitor and should change to CapabilitiesFilterSplitter